server/repository/src/db_diesel/activity_log_row.rs

```diff
@@ -104,3 +106,34 @@ impl<'a> ActivityLogRowRepository<'a> {
         Ok(result)
     }
 }
+
+impl Upsert for ActivityLogRow {
+    fn upsert_sync(&self, con: &StorageConnection) -> Result<(), RepositoryError> {
+        ActivityLogRowRepository::new(con).insert_one(self)
```

Comment:
I think you mention this before, and I was opposed to it because for some reason I thought this impl would take the diesel call. This works nicely though!

Reply:
Thanks, I think the idea behind Upsert trait was that it can be used in other places, mock data etc.. and technically can do 

`RecordRow {..}.upsert(con);`

Which in some cases looks more convenient then `RecordRepository::new(con).upsert(RecordRow{..})`.

Although above is subjective, and repository is easier to use in some places then `.upsert`.

The main reason it was done here though is to avoid having logic spread out over multiple files (which are quite far apart)